filter_plugins/group_hosts.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+from ansible import errors
+import jinja2
+import re
+
+pattern = re.compile("(.*\D(?=\d))(\d+)")
+
+def _get_hostvar(context, var_name, inventory_hostname=None):
+    if inventory_hostname is None:
+        namespace = context
+    else:
+        if inventory_hostname not in context['hostvars']:
+            raise errors.AnsibleFilterError(
+                "Inventory hostname '%s' not in hostvars" % inventory_hostname)
+        namespace = context["hostvars"][inventory_hostname]
+    return namespace.get(var_name)
+
+@jinja2.contextfilter
+def group_hosts(context, var_name, inventory_hostname=None):
+    return {g:_group_hosts(context["groups"][g]) for g in var_name}
```

Comment:
this will lead to the output being in a different order to the input. since what is being returned is a dict, can't the user simply do a dict lookup when the result is returned?

Reply:
Hmm, you're right, I missed that it was a dict.